cfgen/cfgen

```diff
@@ -1254,20 +1259,17 @@ def pool_drives(m0conf: Dict[Oid, Any],
     return [drive_id for drive_id, _ in drives]
 
 
-def cluster_encls(m0conf: Dict[Oid, Any]) -> List[Oid]:
-    encls: List[Oid] = []
-    for item in m0conf:
-        if item.type is ObjT.enclosure:
-            encls.append(item)
-    return encls
+def cluster_encls(m0conf: Dict[Oid, Any], svc_t) -> List[Oid]:
```

Comment:
This will create multiple entries of the same enclosure in case of multiple ioservices (and CAS services) per enclosure.
Thus in case of a single enclosure, number of parity_units will still be 1 and tolerance vector for dix and md pools will be invalid  (`  , types.Obj.Pver { id = utils.zoid types.ObjT.Pver 79, data_units = 1, parity_units = 1, spare_units = 0, pool_width = 2, tolerance = [0, 0, 1, 1, 1], sitevs = [ utils.zoid types.ObjT.Objv 84 ] }`) as per this,
```
    def gen_md_tolerance(cls, m0conf: Dict[Oid, Any],
                         pool_desc: Dict[str, Any]) -> Failures:
        encls_nr = len(cluster_encls(m0conf, SvcT.M0_CST_IOS))
        ctrls_per_encl = floor(len(cluster_ctrls(m0conf)) / encls_nr)
        encl_failures = encls_nr - 1
        ctrl_failures = min(encl_failures, (ctrls_per_encl * encl_failures))
        return Failures(0, 0, encl_failures, ctrl_failures, encl_failures)
```

Reply:
Ouch, yes, good catch! Let me fix that...

Reply:
Fixed. Thanks!